### PR TITLE
Fix Codeowners errors

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Description of changes

Remove people from the codeowners file that do not have write access to the repo, though ignored by github.

I am fully open for suggestions to fill those empty places or other solutions that make the codeowner file check pass.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by infinisil** [comment]
@jtojnar See https://github.com/NixOS/nixpkgs/blob/master/.github/CODEOWNERS.

But it doesn't seem like having these extra entries is a functional problem, the other people still get pinged, e.g. in https://github.com/NixOS/nixpkgs/pull/218776#event-8624994619 me and Eelco got pinged (and not nbp).

**Comment by SuperSandro2000** [comment]
> What errors does it cause?

see the banner in https://github.com/NixOS/nixpkgs/blob/master/.github/CODEOWNERS

> so perhaps it will be fixed eventually.

Maybe, eventually.

> But it doesn't seem like having these extra entries is a functional problem

It creates a reviewing problem. It is way easier to verify that the codeowners file is correct if it is green on master and any change can be verified that it is valid, by just looking at the banner. Right now you need to filter out the already none working entries and if something is overlooked it creates confusion why things are not working and a false sense of security.

> and not nbp

Yes, so we should clean it up, so that no one expects that nbp is pinged.

**Review by infinisil — APPROVED** [review]
Sounds good to me

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -66,22 +66,22 @@
...
+/nixos/doc/manual/configuration/modularity.xml
+/nixos/doc/manual/development/assertions.xml
+/nixos/doc/manual/development/meta-attributes.xml
+/nixos/doc/manual/development/option-declarations.xml
+/nixos/doc/manual/development/option-def.xml
+/nixos/doc/manual/development/option-types.xml
+/nixos/doc/manual/development/replace-modules.xml
+/nixos/doc/manual/development/writing-modules.xml
+/nixos/doc/manual/man-nixos-option.xml
+/nixos/modules/installer/tools/nixos-option.sh
```
Comment (alyssais) [inline .github/CODEOWNERS:84]: Should we just drop these lines?

Comment (SuperSandro2000) [reply .github/CODEOWNERS:84]: Technically they do not do anything, so we could.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -134,7 +134,7 @@
 /pkgs/development/ruby-modules      @marsam
 
 # Rust
-/pkgs/development/compilers/rust @Mic92 @LnL7 @zowoq @winterqt @figsoda
+/pkgs/development/compilers/rust @Mic92 @zowoq @winterqt @figsoda
```
Comment (yurrriq) [inline .github/CODEOWNERS:137]: I'm surprised that @LnL7 doesn't have write access. Should we grant that instead of removing here?

Comment (SuperSandro2000) [reply .github/CODEOWNERS]: That commit is already on master after a rebase, so this not really up for discussion here.

File: .github/CODEOWNERS (hunk 1 above)
Comment (SuperSandro2000) [reply .github/CODEOWNERS]: I just dropped them. Thanks for the reminder.

**Pull request merged**